torchsynth/synth.py

```diff
@@ -172,6 +176,8 @@ def forward(
         """
         if batch_idx is not None:
             self.randomize(seed=batch_idx)
+        else:
+            assert not self.synthconfig.reproducible
```

Comment:
If we are going to assert this can we raise an error with a message here: "Reproducible mode is on, you must pass a batch index when calling this synth"